fix: support new path to CUDAGeneratorImpl.h

shintaro-iwasaki opened the pull request:
OSS PyTorch moved `ATen/CUDAGeneratorImpl.h` to `ATen/cuda/CUDAGeneratorImpl.h` in https://github.com/pytorch/pytorch/pull/70650. This patch makes apex use the new path if a newer PyTorch package is used.

(This is my first time to create a PR for this repo. Please teach me any rule to send a patch)

crcrpar commented:
thanks for the PR but could you apply the same change to some more files which include CUDAGeneratorImpl.h (see: https://github.com/NVIDIA/apex/search?q=CUDAGeneratorImpl)?

shintaro-iwasaki commented:
Thanks, @crcrpar. I overlooked those.

Can I just rewrite the header path to `ATen/cuda/`? Or should I use a macro like `OLD_GENERATOR` to support both old and new PyTorch?

crcrpar commented:
I like the current approach, could you please keep the current approach for apex to be somewhat backward compatible?

shintaro-iwasaki commented:
@crcrpar: Thanks! I updated the commit by following the current approach.

Since `ATen/CUDAGenerator.h` is no longer used (they were changed 2 years ago and, for example, `apex/contrib/csrc/multihead_attn/softmax.cuh` directly uses `ATen/CUDAGeneratorImpl.h`), I removed the corresponding code in `apex/contrib/csrc/multihead_attn/dropout.cuh`.

crcrpar reviewed (approved):
Much appreciated.
